plugins/modules/idrac_support_assist.py

```diff
@@ -0,0 +1,712 @@
+      - System Information will be available in collection by default.
+      - C(hardware_data) The SupportAssist collection will include data related to hardware.
+      - C(storage_logs) The SupportAssist collection will include logs related to storage devices.
+      - C(os_and_app_data) The SupportAssist collection will include data related to operating system and applications.
+      - C(debug_logs) The SupportAssist collection will include logs related to debug.
+      - C(telemetry_reports) The SupportAssist collection will include reports related to telemetry.
+      - C(gpu_logs) The SupportAssist collection will include logs related to GPUs.
+    type: list
+    elements: str
+    choices: [hardware_data, storage_logs, os_app_data, debug_logs, telemetry_reports, gpu_logs]
+  job_wait:
+    description:
+      - Provides the option to wait for job completion.
+    type: bool
+    default: true
+  job_wait_timeout:
+    description:
+      - Time in seconds to wait for job completion.
+      - This is applicable when I(job_wait) is C(true).
+    type: int
+    default: 3600
+  share_parameters:
+    description:
+      - Parameters that are required for the export operation of support assist collection.
+      - I(share_parameters) is required when I(export) is C(true).
+    type: dict
+    suboptions:
+      share_type:
+        description:
+          - Share type of the network share.
+          - C(local) uses local path for I(export) operation.
+          - C(nfs) uses NFS share for I(export) operation.
+          - C(cifs) uses CIFS share for I(export) operation.
+          - C(http) uses HTTP share for I(export) operation.
+          - C(https) uses HTTPS share for I(export) operation.
+          - C(ftp) uses FTP share for I(export) operation.
+        type: str
+        choices: [local, nfs, cifs, http, https, ftp]
+        default: local
+      ip_address:
+        description:
+          - IP address of the network share.
+          - I(ip_address) is required when I(share_type) is C(nfs), C(cifs), C(http) or C(https).
+          - I(ip_address) is not required when I(share_type) is C(local).
+        type: str
+      share_name:
+        description:
+          - Network share path or full local path of the directory for exporting the SupportAssist collection file.
+          - The default path will be current directory when I(share_type) is C(local)
+        type: str
+      workgroup:
+        description:
+          - Workgroup of the network share.
+          - I(workgroup) is applicable only when I(share_type) is C(cifs).
+        type: str
+      username:
+        description:
+          - Username of the network share.
+          - I(username) is required when I(share_type) is C(cifs).
+        type: str
+      password:
+        description:
+          - Password of the network share.
+          - I(password) is required when I(share_type) is C(cifs).
+        type: str
+      ignore_certificate_warning:
```

Comment:
can this parameter be boolean?

Reply:
As per the iDRAC GUI and API, it has got these choices. Same was updated in the specs as well.